src/providers/doppler.js

```diff
@@ -3,20 +3,31 @@ const { VERSION } = require("../meta");
 
 /**
  * Fetch secrets from Doppler the API.
- * @param {{dopplerToken: string}} [{dopplerToken: process.env.DOPPLER_TOKEN}] Requires a Doppler Service Token for API authentication. See https://docs.doppler.com/docs/enclave-service-tokens
+ * @param {{dopplerToken: string}} [{dopplerToken: process.env.DOPPLER_TOKEN}] Requires a Doppler Token for API authentication. See https://docs.doppler.com/docs/enclave-service-tokens
+ * @param {{dopplerProject: string}} [{dopplerProject: null}] Optional Doppler Project. Required when using any token type other than Service Tokens.
+ * @param {{dopplerConfig: string}} [{dopplerConfig: null}] Optional Doppler Config. Required when using any token type other than Service Tokens.
  * @returns {() => Promise<Record<string, string>>}
  */
-async function fetch({ dopplerToken = process.env.DOPPLER_TOKEN } = {}) {
+async function fetch({ dopplerToken = process.env.DOPPLER_TOKEN, dopplerProject = null, dopplerConfig = null } = {}) {
   if (!dopplerToken) {
     throw new Error("Doppler API Error: The 'DOPPLER_TOKEN' environment variable is required");
   }
 
+  const tokenParts = dopplerToken.split(".");
+  if (tokenParts[1] !== "st" && (!dopplerProject || !dopplerConfig)) {
+    throw new Error("Doppler API Error: The provided Doppler Token requires that a Project and Config be specified");
+  }
+
   return new Promise(function (resolve, reject) {
     const authHeader = `Bearer ${dopplerToken}`;
     const userAgent = `gitops-secrets-nodejs/${VERSION}`;
+    let url = "https://api.doppler.com/v3/configs/config/secrets/download?format=json";
+    if (dopplerProject) {
+      url = `https://api.doppler.com/v3/configs/config/secrets/download?format=json&project=${dopplerProject}&config=${dopplerConfig}`;
```

Comment:
Two callouts here:

- Can we escape `dopplerProject` and `dopplerConfig` here?
- Can we omit `dopplerConfig` from the query string if it isn't provided?

You might be able to knock out both of these by using the [querystring lib](https://nodejs.org/api/querystring.html)

Reply:
I tweaked this so it's getting passed through via `querystring.stringify()`. This does result in empty params being passed though (e.g., `project=example&config=`). From what I can tell, this still works fine in situations where the project and config are inferred. Doing this also allows the logic to remain extremely simple. Any objections to this approach?

Reply:
That's perfect 👍 The API will treat these empty query params the same as if they were omitted